rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp

```diff
@@ -301,12 +302,12 @@ const TypeObject * GetCompleteObject(
   // DDS document)
   eprosima::fastcdr::Cdr ser(
     fastbuffer, eprosima::fastcdr::Cdr::LITTLE_ENDIANNESS,
-    eprosima::fastcdr::Cdr::DDS_CDR);  // Object that serializes the data.
+    eprosima::fastcdr::CdrVersion::XCDRv1);  // Object that serializes the data.
```

Comment:
Should we set the enconding flag here?

Reply:
This is following what is done by Fast DDS Gen [here](https://github.com/eProsima/Fast-DDS-Gen/blob/master/src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectSource.stg#L226-L229), so I don't know.

Reply:
I'd like @richiware's input on this

Reply:
Fast DDS Gen doesn't set the encoding flag because Fast CDR uses PLAIN_CDR as default.

Reply:
Done